ext/spl/spl_fixedarray.c

```diff
@@ -106,15 +106,19 @@ static void spl_fixedarray_resize(spl_fixedarray *array, zend_long size) /* {{{
 
 	/* clearing the array */
 	if (size == 0) {
-		zend_long i;
-
-		for (i = 0; i < array->size; i++) {
-			zval_ptr_dtor(&(array->elements[i]));
-		}
+		if (array->elements != NULL) {
+			zend_long i;
+			zval *elements = array->elements;
 
-		if (array->elements) {
-			efree(array->elements);
 			array->elements = NULL;
+			array->size = 0;
+
+			for (i = 0; i < array->size; i++) {
```

Comment:
```suggestion
			for (i = 0; i < size; i++) {
```

Sorry, I didn't notice that `array->size` would always be 0 (2 lines above this line) and this would do 0 loop iterations when I reordered the lines in my previous suggestion.

Reply:
`array_size` and `size` are both 0 at this point. I think we need to store the `array->size` before setting it to 0.

Reply:
> I think we need to store the array->size before setting it to 0.

Oops, that's what I should have said. I did that elsewhere in the example of what could be done instead, in the file linked in the original bug - https://github.com/TysonAndre/pecl-teds/blob/0.1.0a1/teds_vector.c#L409-L419

```c
zend_long old_size = array->size;
array->size = 0;
```